CRM/Import/Form/DataSource.php

```diff
@@ -176,7 +177,12 @@ protected function processDatasource(): void {
       $this->flushDataSource();
       $this->updateUserJobMetadata('submitted_values', $this->getSubmittedValues());
     }
-    $this->instantiateDataSource();
+    try {
+      $this->instantiateDataSource();
+    }
+    catch (CRM_Core_Exception $e) {
+      CRM_Core_Error::statusBounce($e->getUserMessage());
```

Comment:
@eileenmcnaughton I tried a SQL import with `select * from foo`, and it failed like this:

<img width="830" alt="Screen Shot 2023-02-21 at 2 21 59 PM" src="https://user-images.githubusercontent.com/1336047/220471965-50ace418-d2ae-4e51-926e-a2ec428d4f39.png">

Changing to `getMessage()` worked locally...

Reply:
@totten ah yeah - that fell out when I split - OK - will move that part out of this

Reply:
@totten ok fixed - I'll fix up the follow on PR once this is merged